File: mfga-xposed/app/src/main/java/com/mfga/xposed/FontForceCore.java
package com.mfga.xposed;

import android.graphics.Typeface;

/** Android 16 text-family replacement, not a universal/native rendering hook. */
public final class FontForceCore {
    private FontForceCore() {}

    public static Typeface systemReplacementFor(Typeface original) {
        return ReplacementGuard.replace(original, () ->
                Typeface.create(Typeface.DEFAULT, original.getWeight(), original.isItalic()));
    }
}

File: mfga-xposed/app/src/main/kotlin/com/mfga/xposed/FontForceCore.kt
package com.mfga.xposed

import android.graphics.Typeface

/** Android 16 text-family replacement, not a universal/native rendering hook. */
object FontForceCore {
    @JvmStatic
    fun systemReplacementFor(original: Typeface): Typeface? =
        ReplacementGuard.replace(original) {
            Typeface.create(Typeface.DEFAULT, original.weight, original.isItalic)
        }
}
